auth-service/src/main/java/com/auth/authservice/security/UserPrincipalImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.auth.authservice.security;
+
+import com.auth.authservice.datalayer.user.User;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+public class UserPrincipalImpl implements UserDetails {
```

Comment:
What is user principle?

Reply:
It's just the users information that is pertinent to login and authentication as-well as authorization.